src/tactic/simpa.lean

```diff
@@ -24,18 +24,19 @@ This is a "finishing" tactic modification of `simp`. It has two forms.
 * `simpa [rules, ...]` will simplify the goal and the type of a
   hypothesis `this` if present in the context, then try to close the goal using
   the `assumption` tactic. -/
-meta def simpa (use_iota_eqn : parse $ (tk "!")?) (no_dflt : parse only_flag)
-  (hs : parse simp_arg_list) (attr_names : parse with_ident_list)
+meta def simpa (use_iota_eqn : parse $ (tk "!")?) (trace_lemmas : parse $ (tk "?")?)
+  (no_dflt : parse only_flag) (hs : parse simp_arg_list) (attr_names : parse with_ident_list)
   (tgt : parse (tk "using" *> texpr)?) (cfg : simp_config_ext := {}) : tactic unit :=
 let simp_at lc (close_tac : tactic unit) := focus1 $
-  simp use_iota_eqn no_dflt hs attr_names (loc.ns lc) {fail_if_unchanged := ff, ..cfg} >>
+  simp use_iota_eqn trace_lemmas no_dflt hs attr_names (loc.ns lc)
+    {fail_if_unchanged := ff, ..cfg} >>
   (((close_tac <|> trivial) >> done) <|> fail "simpa failed") in
 match tgt with
 | none := get_local `this >> simp_at [some `this, none] assumption <|> simp_at [none] assumption
 | some e := focus1 $ do
   e ← i_to_expr e <|> do {
     ty ← target,
-    e ← i_to_expr_strict ``(%%e : %%ty), -- for positional error messages, don't care about the result
+    e ← i_to_expr_strict ``(%%e : %%ty), -- for positional error messages, don't care about result
```

Comment:
```suggestion
    e ← i_to_expr_strict ``(%%e : %%ty), -- for positional error messages, we don't care about the result
```
the grammar was slightly worsened here?

Reply:
the line length was shortened though...

Reply:
should I move the comment to a new line?

Reply:
How about putting it on the previous line, like this?
```suggestion
    -- for positional error messages, we don't care about the result
    e ← i_to_expr_strict ``(%%e : %%ty),
```